src/api/profile/profile.api.js

```diff
@@ -0,0 +1,118 @@
+                url: `${API_SERVICE_ALTI_BASE_URL}rest/services/profile.json`,
+                method: 'POST',
+                params: {
+                    offset: 0,
+                    sr: CoordinateSystems.LV95.epsgNumber,
+                    distinct_points: true,
+                },
+                data: {
+                    type: 'LineString',
+                    coordinates: chunk.coordinates,
+                },
+            })
+            // checking validity of data from the backend, there is small edge case where we request the backend
+            // because the coordinates are technically in bound of LV95, but our backend doesn't cover the last km
+            // or so of the LV95 bounds, resulting in an empty profile being sent by the backend even though our
+            // coordinates were inbound (hence the dataForChunk.data.length > 2 check)
+            if (dataForChunk && dataForChunk.data && dataForChunk.data.length > 2) {
+                return parseProfileFromBackendResponse(dataForChunk.data, startingDist)
+            } else {
+                log.error('Incorrect/empty response while getting profile', dataForChunk)
+            }
+        } catch (err) {
+            log.error('Error while trying to fetch profile data', err)
+        }
+    }
+    // returning a chunk without data (and also evaluating distance between point as if we were on a flat plane)
+    let lastDist = startingDist
+    let lastCoordinate = startingPoint
+    return new ElevationProfileSegment([
+        ...chunk.coordinates?.map((coordinate) => {
+            let dist = lastDist
+            if (lastCoordinate) {
+                dist += Math.sqrt(
+                    Math.pow(lastCoordinate[0] - coordinate[0], 2) +
+                        Math.pow(lastCoordinate[1] - coordinate[1], 2)
+                )
+            }
+            lastDist = dist
+            lastCoordinate = coordinate
+            return new ElevationProfilePoint(coordinate, round(dist, 2))
+        }),
+    ])
+}
+
+/**
+ * @param {CoordinatesChunk[]} chunks
+ * @returns {CoordinatesChunk[]}
+ */
+function chunksToLV95(chunks) {
+    return chunks.map((chunk) => {
+        chunk.coordinates = toLv95(chunk.coordinates, CoordinateSystems.WEBMERCATOR.epsg)
+        return chunk
+    })
+}
+
+/**
+ * Gets profile from https://api3.geo.admin.ch/services/sdiservices.html#profile
+ *
+ * @param {[Number, Number][]} coordinates Coordinates in EPSG:3857 (WebMercator) from which we want
+ *   the profile
+ * @returns {ElevationProfile}
+ */
+export default async (coordinates) => {
+    if (!coordinates || coordinates.length === 0) {
+        const errorMessage = `Coordinates not provided`
+        log.error(errorMessage)
+        throw errorMessage
+    }
+    const segments = []
+    const coordinateChunks = splitIfOutOfLV95Bounds(coordinates)
+    if (!coordinateChunks) {
+        log.error('No chunks found, no profile data could be fetched')
+        return null
```

Comment:
or should not be an empty segment here ?

Reply:
If I understand it correctly if the request is totally out of switzerland this method return null, and if it is in switzerland but the backend is failing it return an empty segment ? Seems not very consistent.

Reply:
as the function I've written to separate in chunks can return null, whenever the input is malformed or invalid, I wanted to cover this case so that it wouldn't raise an unwanted exception (or console error that would break all the script) later on.

It should NEVER happen, but as the function can return null, well... let's cover it in case. That was why I've written this if block